### DISCUSSION TITLE
Does DEL command delete all the elements of list/set/zset/etc ?

### DISCUSSION THREAD

**Discussion by torwig** [apache/kvrocks#604] [Q&A] [answered]
According to https://github.com/apache/incubator-kvrocks/blob/unstable/src/redis_db.cc#L117 : 

```
rocksdb::Status Database::Del(const Slice &user_key) {
  std::string ns_key;
  AppendNamespacePrefix(user_key, &ns_key);

  std::string value;
  LockGuard guard(storage_->GetLockManager(), ns_key);
  rocksdb::Status s = db_->Get(rocksdb::ReadOptions(), metadata_cf_handle_, ns_key, &value);
  if (!s.ok()) return s;
  Metadata metadata(kRedisNone, false);
  metadata.Decode(value);
  if (metadata.Expired()) {
    return rocksdb::Status::NotFound(kErrMsgKeyExpired);
  }
  return storage_->Delete(rocksdb::WriteOptions(), metadata_cf_handle_, ns_key);
}
```

`DEL` command only deletes metadata of the list/set/zset. 
What about the elements of the deleted list/set/zset? Do they stay in the storage?

**Comment by git-hulk** [comment, marked as answer]
Those deleted elements would be recycled when compacting: https://github.com/apache/incubator-kvrocks/blob/unstable/src/compact_filter.cc#L84

**Reply by torwig** [reply]
If I run
`rpush list1 a1 a2 a3 a4 a5 a6 a7`
`del list1`
`rpush list1 b1 b2 b3`
then due to the compaction filter elements `a1`..`a7` will be recycled?

**Reply by git-hulk** [reply]
Yes, exactly.

**Reply by torwig** [reply]
Thank you.
